materialize-snowflake/snowflake.go

```diff
@@ -518,29 +523,63 @@ func (d *transactor) Load(it *pm.LoadIterator, loaded func(int, json.RawMessage)
 	if len(subqueries) == 0 {
 		return nil // Nothing to load.
 	}
-	var loadAllSQL = strings.Join(subqueries, "\nUNION ALL\n") + ";"
 
-	// Issue a join of the target table and (now staged) load keys,
-	// and send results to the |loaded| callback.
-	rows, err := d.load.conn.QueryContext(sf.WithStreamDownloader(ctx), loadAllSQL)
+	group, groupCtx := errgroup.WithContext(ctx)
+	group.SetLimit(MaxConcurrentLoads)
+	// Used to ensure we have no data-interleaving when calling |loaded|
+	var mutex sync.Mutex
+
+	// In order for the concurrent requests below to be actually run concurrently we need
+	// a separate connection for each
+	db, err := stdsql.Open("snowflake", d.cfg.ToURI(d.tenant))
```

Comment:
I'd suggest creating a `*db` in `newTransactor`, and having the `*db` reference be a property of `transactor`, basically taking the place of `tenant` as a property. If that's not possible for some reason, we'll at least want to do a deferred close of the db here.